crates/infra/utils/src/cargo/workspace.rs

```diff
@@ -99,12 +99,21 @@ impl CargoWorkspace {
     }
 
     pub fn get_command(subcommand: impl AsRef<str>) -> Result<Command> {
+        let subcommand = subcommand.as_ref();
+
         let mut command = Command::new("cargo")
-            .arg(subcommand.as_ref())
+            .arg(subcommand)
             .flag("--all")
-            .flag("--all-targets")
             .flag("--all-features");
 
+        if subcommand == "doc" {
+            // Rustdoc does not support '--all-targets' and it makes sense to only check default --lib/--bins anyway.
+            // Since this is primarily used to check our own documentation, skip documenting dependencies.
+            command = command.flag("--no-deps");
+        } else {
+            command = command.flag("--all-targets");
+        }
+
         if GitHub::is_running_in_ci() {
             // Using `$RUSTFLAGS' or '--' overrides any rustflags from `.cargo/config.toml'.
             // Using this syntax instead, as it is concatenated with the existing flags:
```

Comment:
I don't think `cargo doc` supports `--config deny warnings`. But regardless, we have to use it, in order not to invalidate the build cache.

But to actually deny on warnings, I wonder if there is an alternative?

Reply:
It uses a separate `rustdocflags`, good catch. I decided to specify it here as well (even if it's not technically used by commands like `cargo check`). I decided to not split it to only pass it when running cargo doc, since at that point there wouldn't be a point to have the `get_command` helper and it'd be better to always separately specify these commands.

Reply:
I think we have to do that anyways, in order not to invalidate the build cache. Thank you!